cmd/metrics.go

```diff
@@ -202,4 +203,26 @@ func metricsHandler() http.Handler {
 				ErrorHandling: promhttp.ContinueOnError,
 			}),
 	)
+
+}
+
+// AuthMiddleware checks if the bearer token is valid and authorized.
+func AuthMiddleware() func(http.Handler) http.Handler {
+	return func(h http.Handler) http.Handler {
```

Comment:
It would be much better if we get rid of those extraneous function definition layers created unnecessarily.

```
func AuthMiddleware(h http.Handler) http.Handler {
	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		config := globalServerConfig.GetPrometheusConfig()
		if !config.Enable {
			w.WriteHeader(http.StatusBadRequest)
			return
		}
		if config.Auth == globalPrometheusJwtAuth {
			_, _, authErr := webRequestAuthenticate(r)
			if authErr != nil {
				w.WriteHeader(http.StatusBadRequest)
				return
			}
		}
		h.ServeHTTP(w, r)
	})
}```

Reply:
makes sense :+1: Thanks